contracts/token/ERC20/EncryptedERC20.sol

```diff
@@ -1,201 +1,182 @@
 // SPDX-License-Identifier: BSD-3-Clause-Clear
-
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.24;
 
 import "fhevm/lib/TFHE.sol";
-import "fhevm/abstracts/Reencrypt.sol";
-import "../../utils/EncryptedErrors.sol";
+import { IEncryptedERC20 } from "./IEncryptedERC20.sol";
 
-abstract contract EncryptedERC20 is Reencrypt, EncryptedErrors {
-    enum ErrorCodes {
-        NO_ERROR,
-        UNSUFFICIENT_BALANCE,
-        UNSUFFICIENT_APPROVAL
-    }
+/**
+ * @title       EncryptedERC20
+ */
+abstract contract EncryptedERC20 is IEncryptedERC20 {
+    /// @notice Number of decimals.
+    uint8 internal constant _DECIMALS = 6;
 
-    struct AllowedErrorReencryption {
-        address spender; // account's address allowed to reencrypt errorCode
-        euint8 errorCode;
-    }
+    /// @notice Total supply.
+    uint64 internal _totalSupply;
 
-    event Transfer(uint256 indexed transferId, address indexed from, address indexed to);
-    event Approval(address indexed owner, address indexed spender);
-    event Mint(address indexed to, uint64 amount);
+    /// @notice Name.
+    string internal _name;
 
-    uint64 internal _totalSupply;
-    string private _name;
-    string private _symbol;
-    uint8 public constant decimals = 6;
+    /// @notice Symbol.
+    string internal _symbol;
 
-    // A mapping from transferId to the AllowedErrorReencryption.
-    mapping(uint256 => AllowedErrorReencryption) internal allowedErrorReencryptions;
+    /// @notice A mapping from `account` address to an encrypted `balance`.
+    mapping(address account => euint64 balance) internal _balances;
 
-    // A mapping from address to an encrypted balance.
-    mapping(address => euint64) internal balances;
+    /// @notice A mapping of the form mapping(account => mapping(spender => allowance)).
+    mapping(address account => mapping(address spender => euint64 allowance)) internal _allowances;
 
-    // A mapping of the form mapping(owner => mapping(spender => allowance)).
-    mapping(address => mapping(address => euint64)) internal allowances;
+    /**
+     * @notice Error when the `sender` is not allowed to access a value.
+     */
+    error TFHESenderNotAllowed();
 
-    constructor(string memory name_, string memory symbol_) EncryptedErrors(uint8(type(ErrorCodes).max)) {
+    /**
+     * @param name_ Name of the token.
+     * @param symbol_ Symbol.
+     */
+    constructor(string memory name_, string memory symbol_) {
         _name = name_;
         _symbol = symbol_;
     }
 
-    // Returns the name of the token.
-    function name() public view virtual returns (string memory) {
-        return _name;
+    /**
+     * @notice See {IEncryptedERC20-approve}.
```

Comment:
I know OpenZeppelin did the same, but i am not a fan of putting the natspecs in the interfact then putting just a reference here. Imo move the original natspec here, it makes things clearer and probably easier for the dev tools which rely on natspec.